src/async_fuse/memfs/kv_engine.rs

```diff
@@ -20,66 +23,127 @@ pub enum ValueType {
     INum(INum),
     /// FileAttr
     Attr(SerialFileAttr),
+    /// Next id allocate range begin
+    NextIdAllocateRangeBegin(u64),
 }
 
 impl ValueType {
-    #[allow(dead_code)]
+    #[allow(dead_code, clippy::wildcard_enum_match_arm)]
     /// Turn the `ValueType` into `SerialNode` then into `S3Node`.
-    //// # Panics
+    /// # Panics
     /// Panics if `ValueType` is not `ValueType::Node`.
     pub async fn into_s3_node<S: S3BackEnd + Send + Sync + 'static, K: KVEngine + 'static>(
         self,
         meta: &S3MetaData<S, K>,
     ) -> S3Node<S> {
         match self {
             ValueType::Node(node) => S3Node::from_serial_node(node, meta).await,
-            ValueType::DirEntry(_) | ValueType::INum(_) | ValueType::Attr(_) => {
+            _ => {
```

Comment:
We'd better list all the possible values. If we add new enum option later, we may handle it incorrectly as there's no error reporting for the wildcard_enum_match.

Reply:
These functions are for turning a common type (ValueType) into one certained type (S3_Node), if later we implement more types, would that be too long or weird? And actually we always are certain a key type is binded with a value type.

Reply:
Added comment for this.